microraiden/microraiden/test/test_http_client.py

```diff
@@ -536,3 +539,49 @@ def test_requests(
         response = microraiden.requests.get(url)
 
     assert response.text == 'success'
+
+
+def test_cooperative_close_denied(
+        default_http_client: DefaultHTTPClient,
+        api_endpoint_address: str,
+        token_address: str,
+        channel_manager_address: str,
+        receiver_address: str
+):
+    cooperative_close_denied_mock = mock.patch.object(
+        default_http_client,
+        'on_cooperative_close_denied',
+        wraps=default_http_client.on_cooperative_close_denied
+    ).start()
+
+    with requests_mock.mock() as server_mock:
+        headers = {
+            HTTPHeaders.TOKEN_ADDRESS: token_address,
+            HTTPHeaders.CONTRACT_ADDRESS: channel_manager_address,
+            HTTPHeaders.RECEIVER_ADDRESS: receiver_address,
+            HTTPHeaders.PRICE: '3'
+        }
+        headers = [headers.copy() for i in range(2)]
+
+        i = 0
```

Comment:
lolwut?
```
i = 0
i += 1
```

Reply:
Both this and the `range(2)` magic up there is for easy reordering. You can just take one header configuration and move the entire block somewhere else in the header chain within the test. I often use this instead of explicit indices for just this reason.

If it's too confusing I can change it.

Reply:
Might make more sense in the other PR:
https://github.com/raiden-network/microraiden/pull/292/files#diff-6909d3c2a4b37b2ee0887ccc30f8107cR577

Reply:
I see. I still think it's much more readable if you just index directly without using variable. In test it doesn't matter that much of course.